### DISCUSSION TITLE
Applimage for Chitchatter

### DISCUSSION THREAD

**Discussion by MohammadTaseenKhan** [jeremyckahn/chitchatter#43] [Show and tell]
@jeremyckahn 

An applimage for chitchatter.im

![image](https://user-images.githubusercontent.com/85288261/196132722-45506f61-3543-456a-9628-090650c27136.png)

```
File Size : 73.6 MB

Platform used for file sharing : wetransfer.com  => Status: Secured ✅

Expires in 1 week
```
[Download Here](https://we.tl/t-Ngzq3VHjaz)

**Comment by jeremyckahn** [comment]
Amazing! It looks like you're using [Nativefier](https://github.com/nativefier/nativefier) to make the AppImage. I'd like to build something like this into the project. Rather than using Nativifier I think it might be simpler to integrate [electron-builder](https://www.electron.build/) into the build pipeline. That would enable (at least for Linux AppImage and Windows targets) [auto-updating](https://www.electron.build/auto-update). I've set up [something similar for another Create React App-based project](https://github.com/jeremyckahn/farmhand/blob/fefa45a03c22c6dac04c3ff85f72cb1f18fbed4f/.github/workflows/run-release.yml#L114-L133), and it works quite well.

I've just filed https://github.com/jeremyckahn/chitchatter/issues/44 to get this done!

**Reply by MohammadTaseenKhan** [reply]
i used appnativefy to build this appleimage

```bash
appnativefy --name "Chitchatter" --url "https://chitchatter.im" --favicon
```

**Reply by MohammadTaseenKhan** [reply]
### How to install Appnativefy:

The software is available as a NPM package. To install it, firstly open terminal and run command to install dependencies:
```bash
sudo apt install nodejs npm wget
```
Then install the tool via npm command:
```bash
sudo npm install -g appnativefy
```
With Appnativefy, you can run a simple command to turn a website into single executable Appimage:
```bash
appnativefy --name "TYPE_PACKAGE_NAME" --url "URL_ADDRESS"
```
You can add more flags at the end of the command to enable more options, for instance:

-     --services enables to sign-in using services such as Microsoft 365 and Google.
-     --favicon forces to use the favicon of the site while making the AppImage.
-     --widevine adds Widevine support to enable playing DRM enabled content.

    For more, run `appnativefy -h`

**Reply by MohammadTaseenKhan** [reply]
[https://www.npmjs.com/package/appnativefy](https://www.npmjs.com/package/appnativefy)

**Reply by jeremyckahn** [reply]
This is great. Thanks for sharing this @MohammadTaseenKhan! This should tide users over until #44 is shipped. 🙂

**Comment by MohammadTaseenKhan** [comment]
### Example Preview:


<pre><font color="#26A269"><b>taseen@taseen-khan</b></font>:<font color="#12488B"><b>~</b></font>$ appnativefy --name &quot;Facebook&quot; --url &quot;https://www.facebook.com&quot; --services --widevine --favicon
/usr/local/lib/node_modules/appnativefy

ATTENTION: Using the **unofficial** Electron from castLabs 
It implements Google&apos;s Widevine Content Decryption Module (CDM) for DRM-enabled playback. 
Simply abort &amp; re-run without passing the widevine flag to default to 11.3.0

Preparing Electron app...

Converting icons...

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn&apos;t cached yet...
Downloading electron-v11.3.0-wvvmp-linux-x64.zip: [======================================------------------------------------] 51% ETA: 113.3 seconds 



</pre>
